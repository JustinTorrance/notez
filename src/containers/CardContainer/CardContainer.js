import React, { Component } from 'react';
import { fetchNotes } from '../../thunks/fetchNotes';
import { connect } from 'react-redux';
import Card from '../../components/Card/Card';
import PropTypes from 'prop-types';

export class CardContainer extends Component {

  componentDidMount() {
    const url = 'http://localhost:3001/api/v1/notes';
    this.props.fetchNotes(url);
  }

  render() {

    let noteCards = this.props.notes.map(note => {
      return <Card key={note.id} id={note.id} title={note.title} listItems={note.listItems}/>
    });

    return(
      <div>
        {noteCards}
      </div>
    )
  }
}

CardContainer.propTypes = {
  notes: PropTypes.array.isRequired,
  fetchNotes: PropTypes.func.isRequired
};

export const mapStateToProps = (state) => ({
  notes: state.notes
});

export const mapDispatchToProps = (dispatch) => ({
  fetchNotes: (url) => dispatch(fetchNotes(url))
});

export default connect(mapStateToProps, mapDispatchToProps)(CardContainer);