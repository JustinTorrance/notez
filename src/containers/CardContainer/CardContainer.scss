.new-note-button-wrapper {
  display: flex;
  // align-items: center;
  margin: 1rem;
  justify-content: center;
}

.new-note-button {
  border-radius: 4px;
  &:hover {
    background-color: hotpink;
    cursor: pointer;
  }
}