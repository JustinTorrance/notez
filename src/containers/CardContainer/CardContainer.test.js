import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer } from './CardContainer';

describe('CardContainer', () => {
  let wrapper;
  const fetchNotes = jest.fn();
  const mockNotes = [
    {
      id: "1",
      title: "Kim's list",
      listItems: [
        {id: "a", text: "practice mui"},
        {id: "b", text: "sleep"}
      ]},
    {
      id: "2",
      title: "Josh's list",
      listItems: [
        {id: "c", text: "watch the dogz"},
        {id: "d", text: "call Hannah"}
      ]},
      {
        id: "3",
      title: "JT's list",
      listItems: [
        {id: "e", text: "don't miss the bus back to boulder"},
        {id: "f", text: "eat that doughnut"}
      ]}
  ];

  beforeEach(() => {
    wrapper = shallow(<CardContainer notes={mockNotes} fetchNotes={fetchNotes}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});