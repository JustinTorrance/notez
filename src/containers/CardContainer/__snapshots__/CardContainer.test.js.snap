// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
"<div>
  <Connect(Card) id=\\"1\\" title=\\"Kim\\\\'s list\\" listItems={{...}} />
  <Connect(Card) id=\\"2\\" title=\\"Josh\\\\'s list\\" listItems={{...}} />
  <Connect(Card) id=\\"3\\" title=\\"JT\\\\'s list\\" listItems={{...}} />
</div>"
`;
